crates/nu-cli/src/commands/history.rs

```diff
@@ -46,85 +46,83 @@ impl Command for History {
     ) -> Result<PipelineData, ShellError> {
         let head = call.head;
 
-        // todo for sqlite history this command should be an alias to `open ~/.config/nushell/history.sqlite3 | get history`
+        // TODO: for sqlite history this command should be an alias to `open ~/.config/nushell/history.sqlite3 | get history`
         if let Some(config_path) = nu_path::config_dir() {
             let clear = call.has_flag("clear");
             let long = call.has_flag("long");
             let ctrlc = engine_state.ctrlc.clone();
 
-            let mut history_path = config_path;
-            history_path.push("nushell");
-            match engine_state.config.history_file_format {
-                HistoryFileFormat::Sqlite => {
-                    history_path.push("history.sqlite3");
-                }
-                HistoryFileFormat::PlainText => {
-                    history_path.push("history.txt");
-                }
-            }
+            let nushell_dir = config_path.join("nushell");
+            let history_path = nushell_dir.join(match engine_state.config.history_file_format {
+                HistoryFileFormat::Sqlite => "history.sqlite3",
+                HistoryFileFormat::PlainText => "history.txt",
+            });
 
             if clear {
                 let _ = std::fs::remove_file(history_path);
-                // TODO: FIXME also clear the auxiliary files when using sqlite
-                Ok(PipelineData::empty())
-            } else {
-                let history_reader: Option<Box<dyn ReedlineHistory>> =
-                    match engine_state.config.history_file_format {
-                        HistoryFileFormat::Sqlite => SqliteBackedHistory::with_file(history_path)
-                            .map(|inner| {
-                                let boxed: Box<dyn ReedlineHistory> = Box::new(inner);
-                                boxed
-                            })
-                            .ok(),
-
-                        HistoryFileFormat::PlainText => FileBackedHistory::with_file(
-                            engine_state.config.max_history_size as usize,
-                            history_path,
-                        )
+                if let HistoryFileFormat::Sqlite = engine_state.config.history_file_format {
+                    let _ = std::fs::remove_file(nushell_dir.join("history.sqlite3-shm"));
+                    let _ = std::fs::remove_file(nushell_dir.join("history.sqlite3-wal"));
```

Comment:
I'm wondering if these should be "if exist, then remove". I'm wondering what will happen if remove_file is called and hte wal and shm files do not exist.

Reply:
actually i've tried seing what happens when running `history --clear` like this when one of the file is missing... and nothing happens :open_mouth: 

i think what happens is that `std::fs::remove_file` returns an `Err` when the file is missing (maybe more an `std::io::Result`).
this `Err` is caught by `let _ = ...` and thus treated => we do nothing when the file is not there.

it looks to me this is what we want?
not giving an error when one of these files is missing :yum:

Reply:
of course, we can handle these `remove_file`s better, either in this PR or in another.
at least, for now, i think there is no issue :ok_hand:

Reply:
If you've tested it and there is no harm, like you said, i'm ok with it. I just didn't want a panic.

Reply:
Not sure how to reproduce this. I run a local dev build (`target\debug\nu.exe`).

I type some commands, then `history --clear`

but the history does not even clear. The *.sqilte3 files exist in the history path.

Reply:
I can think of 2 things that might cause this.
1. The file is removed and then recreated since a new history item is being written.
2. There's a sharing violation in Windows and it's failing silently.

Reply:
Then potentially we may have another bug but on Windows, yes?

Reply:
> If you've tested it and there is no harm, like you said, i'm ok with it. I just didn't want a panic.
>
yup, i've tried `rm ~/.config/nushell/history.sqlite3-shm` followed by a `history --clear` and it does not crash => the I/O error is *handled*, i.e. completely discarded here because it does not matter if the file did not exist as the goal of removing it in the end is fullfilled :yum:

Reply:
Agree. I tried it again with two shell sessions opened just in case.

Reply:
@amtoine You tried this on Windows? If not, then your test is invalid for Windows since Windows handles sharing violations much differently. I'm now thinking that we maybe shouldn't ignore errors if they're thrown but report them with ShellError.

Reply:
no i haven't but i think we should not report errors here.
the goal of `history --clear` is to clear the history, i.e. remove all the history files for the given history method.

if the files are there, there won't be an I/O error being caught and we can just remove the files :ok_hand: 
if one of the file is not there, there will be an I/O error under the hood because the file is missing... but the file is then already removed, so we can just discard the error and move on to the next file!

imo it would not make sense to see a `ShellError` in Nushell when running `history --clear` if one of the history file is missing, i'd rather want an error if the history could not be cleared :yum:

Reply:
> if the files are there, there won't be an I/O error being caught and we can just remove the files 👌

On Windows, if the files are there, and there is sharing violation, the files will not get removed because they're currently in use. If 1 of the 3 gets removed because it's not in use, maybe unpredictable things could happen. So, the error may not be caught but it also failed to do what you told it to do.

Reply:
okey got it, i'm very not familiar with windows :relieved:

Reply:
then we already had this issue before with the main history file because the error is not handled, only caught and absorbed with `let _ = ...` :thinking: